sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/dialect/api/DialectType.kt

```diff
@@ -7,6 +7,10 @@ internal interface DialectType {
 
   val javaType: TypeName
 
+  fun decode(value: CodeBlock): CodeBlock = value
+
+  fun encode(value: CodeBlock): CodeBlock = value
+
```

Comment:
I opted to have the `decode` and `encode` functions on the `DialectType` itself instead of in `IntermediateType` to move the dialect specific kotlin types knowledge out of `DialectType`, and to allow different dialects to decode or encode their types how they wish (inspired by `ResultSet` in Java).